src/recordhost.cpp

```diff
@@ -262,16 +276,25 @@ void RecordHost::setCurrentWorkingDirectory(const QString& cwd)
             m_cwd = cwd;
             emit currentWorkingDirectoryChanged(cwd);
         }
-        return;
+    } else {
+        if (!m_remoteDevice.checkIfDirectoryExists(cwd)) {
```

Comment:
here and elsewhere: these operations must be async, otherwise we'll freeze the UI on a bad network connection